PocketKit/Sources/PocketKit/Home/HomeCollectionViewSectionProvider.swift

```diff
@@ -18,194 +29,87 @@ class HomeViewControllerSectionProvider {
-            offsetBy: 3,
-            limitedBy: recommendations.endIndex - 1
-        ) ?? recommendations.endIndex - 1
-        let recommendationsToShow = recommendations[1...endIndex]
-
-        let miniCardHeight = recommendationsToShow.map { recommendation -> CGFloat in
-            guard let viewModel = viewModel.viewModel(for: recommendation.objectID) else {
-                return 0
-            }
-
-            return RecommendationCell.miniHeight(
-                viewModel: viewModel,
-                availableWidth: width - spacing
-            )
-        }.max() ?? 0
-
-        let twoUpInner = NSCollectionLayoutGroup.horizontal(
-            layoutSize: NSCollectionLayoutSize(
-                widthDimension: .fractionalWidth(1),
-                heightDimension: .absolute(miniCardHeight)
-            ),
-            subitem: NSCollectionLayoutItem(
-                layoutSize: NSCollectionLayoutSize(
-                    widthDimension: .fractionalWidth(0.5),
-                    heightDimension: .absolute(miniCardHeight)
-                )
-            ),
-            count: 2
-        )
-
-        twoUpInner.supplementaryItems = [
-            NSCollectionLayoutSupplementaryItem(
-                layoutSize: NSCollectionLayoutSize(
-                    widthDimension: .fractionalWidth(1),
-                    heightDimension: .absolute(dividerHeight)
-                ),
-                elementKind: "twoup-divider",
-                containerAnchor: NSCollectionLayoutAnchor(edges: .bottom),
-                itemAnchor: NSCollectionLayoutAnchor(edges: .bottom)
-            )
-        ]
+    
+    func carouselSection(for slateID: NSManagedObjectID, in viewModel: HomeViewModel, width: CGFloat) -> NSCollectionLayoutSection? {
+        let numberOfCarouselItems = viewModel.numberOfCarouselItemsForSlate(with: slateID)
+        guard numberOfCarouselItems > 0 else { return nil }
+        let itemSize = NSCollectionLayoutSize(widthDimension: .fractionalWidth(1.0), heightDimension: .fractionalHeight(1.0))
+        let item = NSCollectionLayoutItem(layoutSize: itemSize)
+        item.contentInsets = NSDirectionalEdgeInsets(top: 0, leading: Constants.margin, bottom: 0, trailing: Constants.margin)
 
-        let numberOfRows = Int((Float(recommendationsToShow.count) / 2).rounded(.up))
-        return (
-            group: NSCollectionLayoutGroup.vertical(
-                layoutSize: NSCollectionLayoutSize(
-                    widthDimension: .fractionalWidth(1),
-                    heightDimension: .absolute((miniCardHeight + dividerHeight) * 2)
-                ),
-                subitem: twoUpInner,
-                count: numberOfRows
-            ),
-            height: miniCardHeight
+        let groupSize = NSCollectionLayoutSize(widthDimension: .fractionalWidth(0.80), heightDimension: .absolute(Constants.carouselHeight))
+        let group = NSCollectionLayoutGroup.horizontal(layoutSize: groupSize, subitems: [item])
```

Comment:
calculating the number of items for both the carousel and recent saves caused a couple of crashes, but using the method where we rely on the data sources vs count seems to work best in both for recent saves / carousel. Happy to pair on this to clarify. 

Alternative that led to issues:

```
let groupSize = NSCollectionLayoutSize(widthDimension: .fractionalWidth(0.8*Double(numberOfRecentSaves)), heightDimension: .absolute(Constants.carouselHeight))
 let group = NSCollectionLayoutGroup.horizontal(layoutSize: groupSize, subitem: item, count: numberOfRecentSaves)
```

Reply:
@jacobsimeon not sure why I can't reply to your message on 

```
I actually think the crash that you were seeing is probably resolved per https://github.com/Pocket/pocket-ios/pull/212. I understand that this works but having a layout group with a different number of items from the datasource's snapshot seems risky. Can you test out using a repeating item now that the other bug has been fixed?
```

The crash still exists after rebasing from develop - will see what's going on, but maybe we can pair on this :)